### PR TITLE
OneHot and CompressedOneHot

### PR DESCRIPTION
**Pull request by ganesh-krishnan**
## What
Implemented two new encoders:

1. OneHot: Performs a standard one hot encoding
2. CompressedOneHot: Performs a Unique encoding and then performs a one hot encoding

## Why
OneHot and CompressedOneHot encodings are needed for many sklearn algorithms

### CODE REVIEW

File: lore/__init__.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 __copyright__ = 'Copyright © 2017, Instacart'
 __credits__ = ['Montana Low', 'Jeremy Stanley', 'Emmanuel Turlay']
 __license__ = 'MIT'
-__version__ = '0.5.27'
+__version__ = '0.5.28'
```
Comment (montanalow) [inline lore/__init__.py]: @sarchak beat you to 28 :)

File: lore/encoders.py [hunk 2]
```diff
@@ -564,6 +563,73 @@ def cardinality(self):
...
+    def reverse_transform(self, data): pass
+
+    def cardinality(self):
+        return self.sequence_length
+
+    def sequence_name(self, i, suffix=''):
+        return (self.name + '_%i' + suffix) % i
+
+
+class CompressedOneHot(Unique):
```
Comment (montanalow) [inline lore/encoders.py]: What do you think about combining `OneHot` as optional output on `Unique`, e.g. if you pass  `Unique(one_hot=True)`. My thinking is that it's better to have 1 class with various options, than 3. i.e. `CompressedOneHot` becomes `Unique(one_hot=True, minimum_occurences=10)`

Comment (ganesh-krishnan) [reply lore/encoders.py]: Personally, I like classes that do a few things and do them well. When the same class does too many things, as a user, it becomes difficult to understand how to get it to do, exactly what I want. Not to mention, that making sure all the options play well with each other is a bit of a pain to the developer as well. My preference would be to instead combine OneHot and CompressedOneHot. That's something I should have already done in the first place.

**Comment by ganesh-krishnan** [comment]
Yup, will add tests.

**Comment by ganesh-krishnan** [comment]
@montanalow Take a look and let me know what you think

**Pull request closed without merging**